src/Interpreters/MutationsInterpreter.cpp

```diff
@@ -349,6 +350,34 @@ static void validateUpdateColumns(
     }
 }
 
+std::pair<bool, std::vector<ASTPtr>> getFullNestedSubColumnUpdatedExpr(
```

Comment:
I am afraid that I cannot using std::optional instead of pair, 
because not matter `result.first` is true or false `result.second` have at least one ASTPtr.